### DISCUSSION TITLE
New release 1.6.0 ETA?

### DISCUSSION THREAD

**Discussion by lahayeg** [tuya/tuya-home-assistant#505] [Q&A] [answered]
Hello @zlinoliver when do you plan to release new version. There is a PR (in master branch but not yet in 1.6.0 branch) that I would love to have!

**Comment by zlinoliver** [comment, marked as answer]
Hi @lahayeg , we are working closely with Home Assistant team to replace the [existing Tuya integration](https://www.home-assistant.io/integrations/tuya/) with our new Tuya v2. Here are the related PR [links](https://github.com/home-assistant/core/pulls?q=tuya). After Tuya v2 is released on Home Assistant website, we will release the v1.6.0 version with all the new changes. It will happen in October soon.

BTW, which PR you want? Please send the link.

**Reply by lahayeg** [reply]
@zlinoliver  Awesome! PR: https://github.com/tuya/tuya-home-assistant/pull/464

**Reply by bellini666** [reply]
@zlinoliver with the v2 integration now being directly provided by home assistant, what are the differences between it and using this repository?

**Reply by burnworld3d** [reply]
> @zlinoliver with the v2 integration now being directly provided by home assistant, what are the differences between it and using this repository?

There is a lot of differences. V2 Beta: https://github.com/tuya/tuya-home-assistant/tree/master/custom_components/tuya_v2 
V2 Official HA: https://github.com/home-assistant/core/tree/dev/homeassistant/components/tuya

A lot of devices types are not supported yet on official V2 tuya one.

@zlinoliver after initial Tuya V2 official HA addon has been released, any plan to give some love to the current Tuya V2 Beta one?
There is more than 30+ PR awaiting to be merged that would make this work a lot better than it's working now.
Same as for the 'Issues' category, currently is a bit forgotten.
Imho effords should go to push this Beta version, and then transfer all this content to the official addon, and no vice versa

regards

**Reply by bellini666** [reply]
Oh, I see. I thought there was an old version of this exactly same code, so this one would be newer and more supported.

So, if I want to have the latest and greatest version I should move to the official one, right? Do you guys know if there are any way to migrate one to another without having to reconfigure everything (not the old to the new, move from this repo to the official one)?

**Reply by jypsie** [reply]
> @zlinoliver with the v2 integration now being directly provided by home assistant, what are the differences between it and using this repository?

1. It is overly complicated to setup, and makes you even more locked into the cloud
2. V2 works for good 10%, a lot of devices are not recognized or do not work (properly). So don't upgrade ! 
3. I now decided to leave it for what it is..... see no reason or added value....

**Reply by bellini666** [reply]
@jypsie your response confused me =P

You said that there's no reason to upgrade to V2, but if I understood correctly, the official integration provided by HA since their latest update is V2, and thus, the same as this repository, right?

I'm already using this repository (V2) through HACS. My question was regarding any code differences between this repository and the one that got merged there, and if is there an easy migration path without having to remove it and do it all over again. Since like you said, the setup is very complicated and I don't want to lose too much time renaming devices/automations/etc

So, I think your reply is much more a V1 -> V2 issue, right?

**Reply by jypsie** [reply]
Indeed, stay with the old version, V2 is nothing but problems, from setup to getting your devices in HA end then working correctly .... and what should we have additionally?????

**Comment by zlinoliver** [comment]
Hi @lahayeg @bellini666 @burnworld3d @jypsie , Tuya v2 and Tuya HA Core cannot be used concurrently, after discussing with the Home Assistant team, in order to provide Home Assistant users a great experience using Tuya products, we decided to stop maintain the Tuya v2 version in this repository and only provide the [Tuya HA Core version](https://www.home-assistant.io/integrations/tuya/). Trying to maintain the Tuya v2 integration alongside the version in Home Assistant Core is a waste of resources for us and providing these two versions at the same time are confused to our users.

The Tuya developer team really appreciates your contributions and efforts to this Tuya v2 integration project. Please help to recreate your pull request or issues in the [Home Assistant Core repository](https://github.com/home-assistant/core/tree/dev/homeassistant/components/tuya).
